src/state/models/discovery/user-autocomplete.ts

```diff
@@ -77,28 +102,40 @@ export class UserAutocompleteModel {
   // internal
   // =
 
-  async _getFollows() {
-    const res = await this.rootStore.agent.getFollows({
-      actor: this.rootStore.me.did || '',
-    })
-    runInAction(() => {
-      this.follows = res.data.follows.filter(f => !isInvalidHandle(f.handle))
-      for (const f of this.follows) {
-        this.knownHandles.add(f.handle)
+  _computeSuggestions(searchRes: AppBskyActorDefs.ProfileViewBasic[] = []) {
+    if (this.prefix) {
+      const items: ProfileViewBasic[] = []
+      for (const item of this.follows) {
+        if (prefixMatch(this.prefix, item)) {
+          items.push(item)
+        }
+        if (items.length >= 8) {
+          break
+        }
       }
-    })
+      for (const item of searchRes) {
+        if (!items.find(item2 => item2.handle === item.handle)) {
+          items.push({
+            did: item.did,
+            handle: item.handle,
+            displayName: item.displayName,
+            avatar: item.avatar,
+          })
+        }
+      }
+      this._suggestions = items
+    } else {
+      this._suggestions = this.follows
+    }
   }
+}
 
-  async _search() {
-    const res = await this.rootStore.agent.searchActorsTypeahead({
-      term: this.prefix,
-      limit: 8,
-    })
-    runInAction(() => {
-      this.searchRes = res.data.actors
-      for (const u of this.searchRes) {
-        this.knownHandles.add(u.handle)
-      }
-    })
+function prefixMatch(prefix: string, info: ProfileViewBasic): boolean {
```

Comment:
Currently `dho` prefix matches `joshdholtz` but I think it shouldn't. I suggest maybe splitting the `info.handle` and `info.displayName` into "words" by separators (spaces and punctuation), then returning `true` if any of the "words" starts with the given prefix (rather than includes).

Reply:
I debated this, figured it's better to match too many than too few. You're selecting from the menu anyway, so the odds that our lack of specificity could be a problem seem low -- eventually you'll type `dholms` and that's unlikely to double match. Meanwhile you're more likely to type `holm` for holmgren and if dan has a nonreal display name, then holm wouldnt match on a startswith

Would probably have to test against a big sample set to give a definitive answer but that's my thinking

Reply:
I guess my concern is more that some matches may look "unexplainable" and push down the matches you actually want. Even if you keep typing, that seems unfortunate.

I see your point for handles but maybe not for display names.

Reply:
Think about how often people do something like this:

`🧛Paul Frazee`

Reply:
Oh interesting, I guess punctuation alone wouldn't be enough. Okay I'd need to see how it feels in practice but point taken

Reply:
We'll keep an eye on it